nltk/sem/relextract.py

```diff
@@ -198,17 +198,20 @@ def extract_rels(subjclass, objclass, doc, corpus='ace', pattern=None, window=10
     :return: see ``mk_reldicts``
     :rtype: list(defaultdict)
     """
-
-    if subjclass and subjclass not in NE_CLASSES[corpus]:
-        if _expand(subjclass) in NE_CLASSES[corpus]:
-            subjclass = _expand(subjclass)
-        else:
-            raise ValueError("your value for the subject type has not been recognized: %s" % subjclass)
-    if objclass and objclass not in NE_CLASSES[corpus]:
-        if _expand(objclass) in NE_CLASSES[corpus]:
-            objclass = _expand(objclass)
-        else:
-            raise ValueError("your value for the object type has not been recognized: %s" % objclass)
+    
+    for i, s in enumerate(subjclass):
```

Comment:
What about something like this, it's still readable and avoids code duplication.

```python
def extract_rels(...):

    def expand(classes):
        for class_ in classes:
            if class_ and class_ is not in NE_CLASSES[corpus]:
                exapnded = _expand(class_)

                if expanded in NE_CLASSES[corpus]
                    yield expanded
                else:
                    raise ValueError("Your value for the type has not been recognized: %s" % o)

    subjclass = set(expand(subjclass))
    objclass = set(expand(objclass))

```